test/e2e/suite_test.go

```diff
@@ -839,18 +839,28 @@ func checkNodeToolStatusUN(
 	)
 }
 
-func checkSecretExists(t *testing.T, f *framework.E2eFramework, ctx context.Context, kcKey client.ObjectKey, secretKey framework.ClusterKey) {
-	secret := getSecret(t, f, ctx, secretKey)
-	require.True(t, labels.IsManagedBy(secret, kcKey), "secret %s/%s is not managed by %s", secretKey.Namespace, secretKey.Name, kcKey.Name)
+func checkSecretManagedBy(t *testing.T, f *framework.E2eFramework, ctx context.Context, kcKey client.ObjectKey, secretKey framework.ClusterKey) {
+	secret := &corev1.Secret{}
+	require.Eventually(
+		t,
+		func() bool {
+			return f.Get(ctx, secretKey, secret) == nil && labels.IsManagedBy(secret, kcKey)
+		},
+		time.Minute,
+		time.Second,
+		"secret %s does not exist or is not managed by %s",
+		secretKey,
+		kcKey,
+	)
 }
 
 // FIXME there is some overlap with E2eFramework.RetrieveDatabaseCredentials()
 func retrieveCredentials(t *testing.T, f *framework.E2eFramework, ctx context.Context, secretKey framework.ClusterKey) (string, string) {
-	secret := getSecret(t, f, ctx, secretKey)
+	secret := checkSecretExists(t, f, ctx, secretKey)
 	return string(secret.Data["username"]), string(secret.Data["password"])
 }
 
-func getSecret(t *testing.T, f *framework.E2eFramework, ctx context.Context, secretKey framework.ClusterKey) *corev1.Secret {
+func checkSecretExists(t *testing.T, f *framework.E2eFramework, ctx context.Context, secretKey framework.ClusterKey) *corev1.Secret {
```

Comment:
Why was this renamed to an existing functionName? Especially since it seems to be used like "getSecret" unlike "checkSecretExists" (which I would expect to return a bool)

Reply:
Because other functions that wait on something with `assert.Eventually` are all called `checkSomething`.

Reply:
But they don't return anything (some of them are named verifySomething also). This one is clearly used to return something instead of checking it (I might even say the eventually is in the wrong place there)

Reply:
We definitely need to settle on and document naming conventions for verification functions. Can we do that separately, especially since this PR may fix some of the test flakes?